graphstorm-processing/graphstorm_processing/config/config_conversion/converter_base.py

```diff
@@ -0,0 +1,118 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+"""
+import abc
+from abc import abstractmethod
+from typing import Any
+
+from .meta_configuration import NodeConfig, EdgeConfig
+
+
+class ConfigConverter(abc.ABC):
+    """The Config base Converter"""
+
+    @staticmethod
+    @abstractmethod
+    def convert_nodes(nodes_entries: list[dict]) -> list[NodeConfig]:
+        """Convert a list of node attributes into a list of `NodeConfig` objects."""
+
+    @staticmethod
+    @abstractmethod
+    def convert_edges(edges_entries: list[dict]) -> list[EdgeConfig]:
+        """Convert a list of edge attributes into a list of `EdgeConfig` objects."""
+
+    def convert_to_gsprocessing(self, input_dictionary: dict) -> dict:
+        """
+        Take a graph configuration input dictionary and convert it to a GSProcessing-compatible
+        dictionary.
+
+        The input dict needs to provide two top-level lists-of-dicts for the
+        keys 'nodes' and 'edges'.
+        """
+        # deal with corner case
+        if input_dictionary == {}:
+            return {"version": "v1.0", "graph": {"nodes": [], "edges": []}}
```

Comment:
Why we use v1.0 here, but L46 use gsprocessing-1.0?

Reply:
These were written by different authors, so there's probably some miscommunication. I'll fix this.